client/reader/recent/index.tsx

```diff
@@ -82,9 +84,17 @@ const Recent = ( { viewToggle }: RecentProps ) => {
 				feedId: item.feedId,
 				postId: item.postId,
 			} );
+
+			// Add site icon to feed object so have icon for external feeds
+			if ( post ) {
+				const feedSubscription = getReaderFollowForFeed( state, item.feedId );
+				post.site_icon = feedSubscription?.site_icon;
```

Comment:
should we only do this if a site_icon is not yet set?

Reply:
Fixed https://github.com/Automattic/wp-calypso/pull/100738/commits/ce4c6330b8118b799ccd981dd6f5f057293ddd24